gateways/paypalpayflow/request_builder.go

```diff
@@ -0,0 +1,87 @@
+package paypalpayflow
+
+import (
+	"fmt"
+
+	"github.com/BoltApp/sleet"
+)
+
+var (
+	defaultVerbosity    string = "HIGH"
+	defaultTender       string = "C"
+	defaultMIT          string = "MIT"
+	MITUnscheduled      string = "MITR"
+	CITUnscheduled      string = "CITU"
+	CITInitial          string = "CITI"
+	CITInitialRecurring string = "CITR"
+	MITRecurring        string = "MITR"
+)
+
+func buildAuthorizeParams(request *sleet.AuthorizationRequest) *Request {
+	expirationDate := fmt.Sprintf("%02d%02d", request.CreditCard.ExpirationMonth, request.CreditCard.ExpirationYear%100)
+	amount := sleet.AmountToDecimalString(&request.Amount)
+	var CardOnFile *string = nil
+
+	if request.ProcessingInitiator != nil {
+		switch *request.ProcessingInitiator {
+		case sleet.ProcessingInitiatorTypeInitialRecurring:
+			CardOnFile = &CITInitialRecurring
+		case sleet.ProcessingInitiatorTypeFollowingRecurring:
+			CardOnFile = &CITUnscheduled
```

Comment:
Wondering about this one? This flag is for recurring payments (like a subscription). 

Going off this chart: https://developer.paypal.com/api/nvp-soap/payflow/integration-guide/card-on-file/

Seems like this should be MITR

Reply:
Changed